packages/http/src/types.ts

```diff
@@ -83,3 +75,21 @@ export interface IHttpResponse {
   headers?: IHttpNameValue;
   body?: any;
 }
+
+export type ProblemJson = {
+  name: string;
+  title: string;
+  status: number;
+  detail: string;
+};
+
+export class ProblemJsonError extends Error {
```

Comment:
Can you add a simple unit test showing what we expect ProblemJsonError to look like? E.g. in other tests you do `expectToThrow(ProblemJsonError.fromTemplate(etc))`, but would be nice to see what `ProblemJsonError.fromTemplate(etc)` is meant to be.

Reply:
Isn't this assured by the types?

```ts
public static fromTemplate(template: Omit<ProblemJson, 'detail'>, detail?: string): ProblemJsonError
```

`ProblemJsonError` has `name, message,status,detail`, so there's no way it has *not* such fields. Does that clarify?